src/subplots.ts

```diff
@@ -12,1983 +12,2128 @@ var alert_count = 0;
-            this.refresh_MinMax(graph.point_list, true);
-          }
-          this.nb_graph = this.plotObject.graphs.length;
-        } else if (data['type_'] == 'scatterplot') {
-          this.type_ = 'scatterplot';
-          this.axis_ON = true;
-          this.mergeON = true;
-          this.plotObject = Scatter.deserialize(data);
-          this.plot_datas['value'] = [this.plotObject];
-          this.pointLength = this.plotObject.point_list[0].size;
-          this.scatter_init_points = this.plotObject.point_list;
-          this.refresh_MinMax(this.plotObject.point_list);
-          this.heatmap_view = data["heatmap_view"] || false;
-          if (data["heatmap"]) {this.heatmap = Heatmap.deserialize(data["heatmap"])} else {this.heatmap = new Heatmap();}
-          this.selected_areas = [];
-          for (let i=0; i<this.heatmap.size[0]; i++) {
-            let temp = [];
-            for (let j=0; j<this.heatmap.size[1]; j++) {
-              temp.push(0);
-            }
-            this.selected_areas.push(temp);
-          }
-        }
-        this.isParallelPlot = false;
-        if (this.mergeON && alert_count === 0) {
-          // merge_alert();
-        }
+export class PlotPieChart extends PlotData {
+  public constructor(
+    public data: DataSample[],
+    public width: number,
+    public height: number,
+    public buttons_ON: boolean,
+    public X: number,
+    public Y: number,
+    public canvas_id: string,
+    public is_in_multiplot = false) {
+    super(data, width, height, buttons_ON, X, Y, canvas_id, is_in_multiplot);
+    if (!is_in_multiplot) {
+      let requirement = '0.6.0';
+      check_package_version(data['package_version'], requirement);
     }
-  
-    draw() {
-      this.draw_from_context(false);
-      this.draw_from_context(true);
-    }
-  
-    draw_from_context(hidden) {
-      this.define_context(hidden);
-      this.context.save();
-      this.draw_empty_canvas(this.context);
-      if (this.settings_on) {this.draw_settings_rect();} else {this.draw_rect();}
-      this.context.beginPath();
-      this.context.rect(this.X-1, this.Y-1, this.width+2, this.height+2);
-      this.context.clip();
-      this.context.closePath();
-      this.draw_graph2D(this.plotObject, hidden, this.originX, this.originY);
-      if (this.heatmap_view) {
-        this.draw_heatmap(hidden);
-      } else {
-        this.draw_scatterplot(this.plotObject, hidden, this.originX, this.originY);
-        if (this.permanent_window) {
-          this.draw_selection_rectangle();
-        }
-        if (this.zw_bool || (this.isSelecting && !this.permanent_window)) {
-          this.draw_zoom_rectangle();
+    if (this.buttons_ON) {
+      this.refresh_buttons_coords();
+    }
+    this.log_scale_y = false;
+    this.mergeON = true;
+    this.selected_areas = [];
+    this.refresh_MinMax();
+    this.plotObject = PieChart.deserialize(data);
+    this.plotObject.pieParts = this.plotObject.definePieParts(this.height * 0.475);
```

Comment:
definePieParts could be called directly from the PieChart constructor.
First because it makes sense. 
Second, because this.plotObject typed as `any` and Ctrl+click on definePieParts does not detect this line.

Reply:
Done here https://github.com/Dessia-tech/plot_data/pull/168/commits/0ecc9934412d3825b26d1c38510e641baadf27f2